bids/layout/index.py

```diff
@@ -205,13 +208,13 @@ def _index_dir(self, path, config, force=None):
         if self.config_filename in filenames:
             filenames.remove(self.config_filename)
 
+        all_file_objs = []
         for f in filenames:
             abs_fn = path / f
             # Skip files that fail validation, unless forcibly indexing
             if force or self._validate_file(abs_fn):
-                self._index_file(abs_fn, config_entities)
-
-        self.session.commit()
+                bf, file_objs = self._index_file(abs_fn, config_entities)
+                all_file_objs += file_objs
```

Comment:
Then this would become:

```suggestion
                all_file_objs.extend(self._index_file(abs_fn, config_entities))
```